pwndbg/heap/ptmalloc.py

```diff
@@ -658,17 +658,219 @@ def __init__(self) -> None:
         # ptmalloc cache for current thread
         self._thread_cache: gdb.Value = None
 
-    def largebin_size_range_from_index(self, index):
-        index += NSMALLBINS
-        spaces_table = self._spaces_table()
-        largest_largebin = self.largebin_index(pwndbg.gdblib.arch.ptrmask)
-        start_size = (NSMALLBINS * self.malloc_alignment) - self.malloc_alignment
+        self.largebin_reverse_lookup_32 = [
+            0x200,
+            0x240,
+            0x280,
+            0x2C0,
+            0x300,
+            0x340,
+            0x380,
+            0x3C0,
+            0x400,
+            0x440,
+            0x480,
+            0x4C0,
+            0x500,
+            0x540,
+            0x580,
+            0x5C0,
+            0x600,
+            0x640,
+            0x680,
+            0x6C0,
+            0x700,
+            0x740,
+            0x780,
+            0x7C0,
+            0x800,
+            0x840,
+            0x880,
+            0x8C0,
+            0x900,
+            0x940,
+            0x980,
+            0x9C0,
+            0xA00,
+            0xC00,
+            0xE00,
+            0x1000,
+            0x1200,
+            0x1400,
+            0x1600,
+            0x1800,
+            0x1A00,
+            0x1C00,
+            0x1E00,
+            0x2000,
+            0x2200,
+            0x2400,
+            0x2600,
+            0x2800,
+            0x2A00,
+            0x3000,
+            0x4000,
+            0x5000,
+            0x6000,
+            0x7000,
+            0x8000,
+            0x9000,
+            0xA000,
+            0x10000,
+            0x18000,
+            0x20000,
+            0x28000,
+            0x40000,
+            0x80000,
+        ]
+
+        self.largebin_reverse_lookup_32_big = [
+            0x3F0,
+            0x400,
+            0x440,
+            0x480,
+            0x4C0,
+            0x500,
+            0x540,
+            0x580,
+            0x5C0,
+            0x600,
+            0x640,
+            0x680,
+            0x6C0,
+            0x700,
+            0x740,
+            0x780,
+            0x7C0,
+            0x800,
+            0x840,
+            0x880,
+            0x8C0,
+            0x900,
+            0x940,
+            0x980,
+            0x9C0,
+            0xA00,
+            0xA40,
+            0xA80,
+            0xAC0,
+            0xB00,
+            0xB40,
+            0xB80,  # Bin 95 is unused, but its size is used to calculate the previous bin's maximum chunk size.
```

Comment:
> Probably stupid question

Not at all.

> how is this bin 95?

This is indeed largebin 31, but GLIBC malloc has no largebins array.
Instead, every doubly-linked bin resides in a single array named `bins`. There are 64 bins preceding largebin 31 (62 smallbins, the unsortedbin and the unused bin 0), so this could technically be considered bin 95.

I've amended the comment to clarify this👍